### PR TITLE
Improve grammar syntax

### LINKED ISSUE #93: Syntactic grammar improvements
- [ ] order of individual sections (rules, inits, etc) should be arbitrary
- [ ] require all parts to be separated by newlines - for example, rules can all be specified inline, which is not desired behaviour and can cause weird syntactic constructs

### PR DESCRIPTION
**Pull request by mopichalova**
Order of individual sections (rules, inits, etc) is arbitrary now to allow more flexibility
- rules must be defined at least once
- each section can be defined more than once

Enforcing and accepting newlines:
- ignoring only `WS_INLINE` instead of all `WS`
- specifying where newlines are allowed

Close #93

### CODE REVIEW

File: eBCSgen/Parsing/ParseBCSL.py [hunk 1]
```diff
@@ -87,13 +87,14 @@ def to_side(self):
 
 
 GRAMMAR = r"""
-    model: rules (inits)? (definitions)? (complexes)? (regulation)?
+    model: (sections)* rules (sections | rules)*
```
Comment (xtrojak) [inline eBCSgen/Parsing/ParseBCSL.py:90]: What is the reason for having `rules` there for the second time?

Comment (mopichalova) [reply eBCSgen/Parsing/ParseBCSL.py:90]: To allow `rules` to be defined multiple times, just like `sections`. This keeps things consistent and fair."

File: eBCSgen/Parsing/ParseBCSL.py [hunk 2]
```diff
@@ -593,14 +595,27 @@ def param(self, matches):
...
                 key, value = list(match.items())[0]
                 if key == 'rules':
-                    rules = set(value)
+                    rules.update(value)
+                if key == 'inits':
+                    inits = value
+                if key == 'definitions':
+                    definitions = value
+                if key == 'regulation':
+                    regulation = value
```
Comment (xtrojak) [inline eBCSgen/Parsing/ParseBCSL.py:612]: Maybe there should be `dict.update` again somewhere?

Comment (mopichalova) [reply eBCSgen/Parsing/ParseBCSL.py]: Yes, `dict.update` solved the issue in commit 8546c4e. 
Commit 1ba38d2 handles multiple regulation definitions by throwing an exception.

File: eBCSgen/Parsing/ParseBCSL.py [hunk 3]
```diff
@@ -586,53 +633,79 @@ def inits(self, matches):
...
+                key, value = list(match.children[0].items())[0]
+                if key == "rules":
+                    rules.update(value)
+                if key == "inits":
+                    inits.update(value)
+                if key == "definitions":
+                    definitions.update(value)
+                if key == "regulation":
+                    if regulation:
+                        raise UnspecifiedParsingError("Multiple regulations")
```
Comment (xtrojak) [inline eBCSgen/Parsing/ParseBCSL.py:672]: I believe this code can be simplified - the condtion influences only how `key, value` are created, the rest is the same in both cases.

**Review by xtrojak — CHANGES_REQUESTED** [review]

File: eBCSgen/Parsing/ParseBCSL.py (hunk 3 above)
Comment (mopichalova) [reply eBCSgen/Parsing/ParseBCSL.py]: Right, I addressed it in commit a3eca18.

**Pull request merged**